src/index.ts

```diff
@@ -0,0 +1,25 @@
+import {deriveMasterSK} from "./key-derivation";
+// @ts-ignore
+import secureRandom from "secure-random";
+import {mnemonicToSeedSync, validateMnemonic} from "bip39";
+import assert from "assert";
+import {HDNode} from "./hdnode";
+import {Buffer} from "buffer";
+
+export function generateRandomBLSSecretKey(entropy?: Buffer): Buffer {
+    let ikm = secureRandom(16, {type: "Buffer"});
+    if(entropy) {
+        ikm = Buffer.concat([entropy, ikm]);
+    }
+    return deriveMasterSK(ikm);
+}
+
+export function fromMnemonicToBlsSecretKey(mnemonic: string, path = "m/12381/60/0/0"): Buffer {
```

Comment:
I don't think there should be a mnemonicToSeed that doesn't do validation. Ie: You shouldn't get a return value from an invalid mnemonic.

Reply:
first line in this function validates mnemonic and you won't get return value